src/Message/Toast.elm

```diff
@@ -1,38 +1,62 @@
-module Message.Toast exposing (Toast, decoder, encode, label)
+module Message.Toast exposing (Toast, encode, decoder, label)
+
+{-| Requesting the host application to show a notification of some sort is
+a common use case for cross-frame messaging. The Message.Toast module defines
+types to support this use case.
+
+@docs Toast, encode, decoder, label
+
+-}
 
 import Json.Decode as Decode exposing (Decoder)
 import Json.Decode.Pipeline exposing (optional, required)
 import Json.Encode as Encode
 import LabeledMessage exposing (expectLabel, withLabel)
 
 
+{-| The Toast type represents a request from a client to a host to display
+a notification. It consists of an optional `title`, a `message` to display
+and a placeholder for custom pass-through data to support additional
+app-specific features. Common use cases for app-specific features are things
+like timed auto-dismissal, notification levels or icons, etc.
+-}
 type alias Toast =
-    { title : String
+    { title : Maybe String
     , message : String
     , custom : Maybe Decode.Value
     }
 
 
+{-| This is the label used to tag toast requests in JSON. Other modules
+should not need to reference it, but it is exposed to force a package version
+bump if it changes.
+-}
 label : String
 label =
     "toastRequest"
 
 
+{-| Decoder for Toast messages, which are expected to be tagged with
+`label`
+-}
 decoder : Decoder Toast
 decoder =
     (Decode.succeed Toast
-        |> optional "title" Decode.string ""
+        |> optional "title" (Decode.maybe Decode.string) Nothing
```

Comment:
The decoder and encoder changed here to make testing the decode/encode cycle easier. The test is a preliminary step in showing how we can maintain backwards compatibility checks when evolving the API.